crud/common/sharding.lua

```diff
@@ -51,4 +53,22 @@ function sharding.tuple_set_and_return_bucket_id(tuple, space, specified_bucket_
     return bucket_id
 end
 
+-- Get sharding key (actually field names) for all spaces in schema
+-- and cache it's value to speedup access.
+function sharding.get_ddl_sharding_key(space_name)
+    if box.space._ddl_sharding_key == nil then
```

Comment:
I wrote some ideas about crud and ddl integration in #27. It looks counter intuitive for me to integrate via the hidden space, while the usual Lua API is present (`ddl.get_schema()`).

Reply:
Yep, discussed it orally and decided to use `ddl.get_schema()`.

```lua
@@ -66,15 +68,17 @@ end
 -- Get sharding key (actually field names) for all spaces in schema
 -- and cache it's value to speedup access.
 function sharding.get_ddl_sharding_key(space_name)
-    if box.space._ddl_sharding_key == nil then
+    if ddl == false then
         return nil
     end
-
-    if sharding_key_cache == nil then
-        sharding_key_cache = box.space._ddl_sharding_key:select{}
+    local schema = ddl.get_schema(space_name)
+    local sharding_key
+    if schema.spaces[space_name] == nil or
+       schema.space_name[space_name].sharding_key == nil then
+        return nil
     end
 
-    return sharding_key_cache[space_name]
+    return schema.space_name[space_name].sharding_key
 end
 
 return sharding
```

Reply:
For the history sake: I changed my mind, see https://github.com/tarantool/crud/issues/27#issuecomment-913665805.